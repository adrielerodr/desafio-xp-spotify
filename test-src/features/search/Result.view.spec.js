import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import handlers from './dependencies/functions';

import WrappedResult, {
  Result
} from '@/features/search/views/Result.view';

describe(`Result view test`, function() {
  it('Should render proper html', function() {
    const results = {
      "artists": [
        {
          "name": "Bruno Mars",
        }
      ],
      "id": "6J84szYCnMfzEcvIcfWMFL",
      "images": [
        {
            "height": 636,
            "url": "https://i.scdn.co/image/0a01bbfac984fe3a55192aceea6c0fe9bf495c09",
            "width": 640
        },
        {
            "height": 298,
            "url": "https://i.scdn.co/image/3d279adc8126da0e4b6f2247a6e1dbfbc559e1ac",
            "width": 300
        },
        {
            "height": 64,
            "url": "https://i.scdn.co/image/96ebe0fc8c5c9a0ee315575935f1700ef8e02a1d",
            "width": 64
        }
      ],
      "name": "Doo-Wops & Hooligans",
    };

    const tree = renderer
      .create(
        <WrappedResult
          key={results.id}
          id={results.id}
          name={results.name}
          artist={results.artists[0]}
          image={results.images[1]}
          redirectAlbum={handlers.getAlbum}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
