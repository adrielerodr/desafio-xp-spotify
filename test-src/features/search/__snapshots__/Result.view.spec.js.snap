// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result view test Should render proper html 1`] = `
<a
  className="card"
  onClick={[Function]}
>
  <img
    alt="Doo-Wops & Hooligans"
    height={198}
    src="https://i.scdn.co/image/3d279adc8126da0e4b6f2247a6e1dbfbc559e1ac"
    width={200}
  />
  <p>
    Doo-Wops & Hooligans
  </p>
  <p
    className="text-grey"
  >
    Bruno Mars
  </p>
</a>
`;
